src/trading/NewOrders.sol

```diff
@@ -0,0 +1,338 @@
+            orders[_orderId].betterOrderId = 0;
+            orders[_orderId].worseOrderId = 0;
+        }
+        return true;
+    }
+
+    function takeOrderLog(IMarket _market, uint8 _orderOutcome, Trading.TradeTypes _orderType, bytes20 _orderId, address _taker, uint256 _makerSharesFilled, uint256 _makerTokensFilled, uint256 _takerSharesFilled, uint256 _takerTokensFilled, uint256 _tradeGroupId) public constant returns (bool) { 
+        uint256 _price = orders[_orderId].fxpPrice;
+        address _maker = orders[_orderId].owner;
+        TakeOrder(_market, _orderOutcome, _orderType, _orderId, _price, _maker, _taker, _makerSharesFilled, _makerTokensFilled, _takerSharesFilled, _takerTokensFilled, _tradeGroupId);
+        return true;
+    }
+
+    function completeSetsLog(address _sender, IMarket _market, Trading.TradeTypes _type, uint256 _fxpAmount, uint256 _numOutcomes, uint256 _marketCreatorFee, uint256 _reportingFee) public constant returns (bool) { 
+        CompleteSets(_sender, _market, _type, _fxpAmount, _numOutcomes, _marketCreatorFee, _reportingFee);
+        return true;
+    }
+
+    function cancelOrderLog(IMarket _market, address _sender, uint256 _fxpPrice, uint256 _fxpAmount, bytes20 _orderId, uint8 _outcome, Trading.TradeTypes _type, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed) public constant returns (bool) { 
+        CancelOrder(_market, _sender, _fxpPrice, _fxpAmount, _orderId, _outcome, _type, _fxpMoneyEscrowed, _fxpSharesEscrowed);
+        return true;
+    }
+
+    function modifyMarketVolume(IMarket _market, uint256 _fxpAmount) internal returns (bool) { 
+        marketOrderData[_market].volume += _fxpAmount;
+        _market.getBranch().getTopics().updatePopularity(_market.getTopic(), _fxpAmount);
+        return true;
+    }
+
+    function setPrice(IMarket _market, uint8 _outcome, uint256 _fxpPrice) internal returns (bool) { 
+        marketOrderData[_market].prices[_outcome] = _fxpPrice;
+        return true;
+    }
+
+    function removeOrderFromList(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) private returns (bool) { 
+        bytes20 _betterOrderId = orders[_orderId].betterOrderId;
+        bytes20 _worseOrderId = orders[_orderId].worseOrderId;
+        if (bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)] == _orderId) { 
+            bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)] = _worseOrderId;
+        }
+        if (worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)] == _orderId) { 
+            worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)] = _betterOrderId;
+        }
+        if (_betterOrderId != 0) { 
+            orders[_betterOrderId].worseOrderId = _worseOrderId;
+        }
+        if (_worseOrderId != 0) { 
+            orders[_worseOrderId].betterOrderId = _betterOrderId;
+        }
+        orders[_orderId].betterOrderId = 0;
+        orders[_orderId].worseOrderId = 0;
```

Comment:
`solc` gives me a warning here indicating that assigning a decimal value to `bytesXX` may not behave as expected.

Reply:
Hmm, how can we set a bytes20 variable to 20-byte array of zeros?  This line in trade.se also seems to be causing an error because we're passing in zeros: `return MAKEORDER.makeOrder(sender, direction, fxpAmount, fxpPrice, market, outcome, 0, 0, tradeGroupID)`.

Reply:
You should test first, but I'm guessing for the case of `0` you could just do `bytes20(0)`.  For any other values I suspect you'll need to do something different but we should never be hard-coding non-zero byte arrays.  You should test and verify that Solidity `bytes20(0)` results in a byte array of 20 bytes, to make sure my assumptions are correct.

Reply:
I think bytes20(0) might work in Solidity, but it is causing errors in Serpent.  For example, I tried changing some lines in `trade.takeBestOrder()` from using 0 to using bytes20(0), and I'm seeing `Invalid argument count or LLL function: (bytes20 0)`

Reply:
`new bytes20(0)` maybe?  I'll try to fool around with it in the sandbox and see if I can figure out how to create a `bytes20`.